src/main/java/com/rybka/service/ExchangeService.java

```diff
@@ -1,100 +1,56 @@
-            baseCurrencyAbbreviation = userBaseCurrencyAbbreviation;
-            targetCurrencyAbbreviation = userTargetCurrencyAbbreviation;
-        } catch (Exception e) {
-            System.out.println("Incorrect exchange data!");
-            System.exit(3);
+    public ExchangeRate loadCurrencyOf(String userBaseCurrency, String userTargetCurrency) throws CurrencyAPICallException, ZeroCurrencyRateException {
+        var currencyRateMap = connectorService.retrieveRates(userBaseCurrency);
+        var currencyRateList = createCurrencyObjectList(currencyRateMap);
+        var currencyRate = retrieveJsonData(currencyRateList, userTargetCurrency);
+        if (currencyRate == 0.0) {
+            throw new ZeroCurrencyRateException("Invalid currency");
         }
+        return new ExchangeRate(userBaseCurrency, userTargetCurrency, currencyRate);
     }
 
-    public void exchange(Double amount) {
-
-        this.amount = amount;
-        this.total = amount * currencyValue;
-    }
-
-    public void getCurrencyObject() {
-        Currency currency = new Currency(baseCurrencyAbbreviation, targetCurrencyAbbreviation, currencyValue, amount, total);
-        HibernateDAO hibernateDAO = new HibernateDAO();
-        hibernateDAO.save(currency);
-        hibernateDAO.showTableRow();
-    }
-
-    public Double getCurrencyValue() {
-        return currencyValue;
-    }
-
-    public void setCurrencyValue(Double currencyValue) {
-        this.currencyValue = currencyValue;
+    public ConvertedCurrency calculateTotal(ExchangeRate exchangeRate, Double amount) {
+        ConvertedCurrency currency = new ConvertedCurrency();
+        currency.setBase(exchangeRate.getBase());
+        currency.setTarget(exchangeRate.getTarget());
+        currency.setRate(exchangeRate.getRate());
+        currency.setAmount(amount);
+        currency.setTotal(currency.getAmount() * currency.getRate());
+        return currency;
     }
 
-    public String getBaseCurrencyAbbreviation() {
-        return baseCurrencyAbbreviation;
-    }
-
-    public void setBaseCurrencyAbbreviation(String baseCurrencyAbbreviation) {
-        this.baseCurrencyAbbreviation = baseCurrencyAbbreviation;
-    }
-
-    public String getTargetCurrencyAbbreviation() {
-        return targetCurrencyAbbreviation;
-    }
-
-    public void setTargetCurrencyAbbreviation(String targetCurrencyAbbreviation) {
-        this.targetCurrencyAbbreviation = targetCurrencyAbbreviation;
-    }
-
-    public Double getAmount() {
-        return amount;
-    }
-
-    public void setAmount(Double amount) {
-        this.amount = amount;
-    }
+    public Double retrieveJsonData(List<Currency> currencies, String userTargetCurrency) {
+        var currencyRate = 0.0;
+        currencyRate = currencies.stream()
+                .filter(element -> userTargetCurrency.equals(element.getName()))
+                .findFirst()
+                .get()
```

Comment:
I would made a little bit changes here. We can avoid check for 0.0 in the method where we call this one. 

Try to use another method then .get(), ex. getOrThrow() we can throw an exception directly here instead do this in the caller method.

Reply:
fixed